diffyscan/utils/encoder.py

```diff
@@ -0,0 +1,105 @@
+from .logger import logger
+
+
+def encode_address(address):
+    number = int(address, 16)
+    formatted_address = f"{number:064x}"
+    return formatted_address
+
+
+def encode_uint256(number):
+    return format(number, "064x")
+
+
+def encode_bytes32(data):
+    return data.replace("0x", "")
+
+
+def encode_bytes(data):
+    bytes_str = data.lstrip("0x")
+    data_length = len(bytes_str) // 2
+    encoded_length = 0
+    if data_length > 0:
+        encoded_length = ((len(bytes_str) - 1) // 64 + 1) * 64
```

Comment:
magic

Reply:
source of magic [here](https://github.com/lidofinance/lido-dao/blob/a9a9ea50b5be7dba33e06d70cc068252ffd01b52/bytecode-verificator/bytecode_verificator.sh#L239)

Reply:
now it seems more readable

Reply:
Can it be commented with a brief explanation of the essence of this operation instead of saying 'taken from there and untested'? :pray:

Reply:
```python
# Calculate the length of the hex-encoded 32-bytes padded data 
# since EVM uses 32-byte (256-bit) words
```